polynote-frontend/polynote/toolbar.js

```diff
@@ -1,58 +1,253 @@
-import { FakeSelect } from './fake_select'
-
-export class Toolbar {
-    constructor(id, actions) {
-        this.element = (id instanceof HTMLElement) ? id : document.getElementById(id);
-        this.buttons = [];
-        for (const selector in actions) {
-            if (actions.hasOwnProperty(selector)) {
-                let action = actions[selector];
-                let button = this.element.querySelector(selector);
-
-                if (button.classList.contains('dropdown')) {
-                    button = new FakeSelect(button);
-                } else {
-                    button.setState = (state) => {
-                        if (state && state !== 'false') {
-                            button.classList.add('active');
+import {button, div, fakeSelectElem, h3, iconButton} from "./tags";
+import {FakeSelect} from "./fake_select";
+import {Cell, CodeCell, TextCell} from "./cell";
+import {LaTeXEditor} from "./latex_editor";
+import {UIEvent, UIEventTarget} from "./ui_event";
+
+export class ToolbarEvent extends UIEvent {
+    constructor(eventId, details) {
+        super(eventId, details || {});
+    }
+}
+
+export class ToolbarUI extends UIEventTarget {
+    constructor() {
+        super();
+        this.notebookToolbar = new NotebookToolbarUI().setEventParent(this);
+        this.cellToolbar = new CellToolbarUI().setEventParent(this);
+        this.codeToolbar = new CodeToolbarUI().setEventParent(this);
+        this.textToolbar = new TextToolbarUI().setEventParent(this);
+        this.settingsToolbar = new SettingsToolbarUI().setEventParent(this);
+
+        this.el = div(['toolbar-container'], [
+            this.notebookToolbar.el,
+            this.cellToolbar.el,
+            this.codeToolbar.el,
+            this.textToolbar.el,
+            this.settingsToolbar.el,
+        ])
+    }
+
+    onContextChanged() {
```

Comment:
How does this method get invoked? Does it? I noticed you also deleted the corresponding CSS... do all the toolbars show all the time now?

Reply:
It was hard for me to find the change in this diff - turns out the ui.js diff wasn't rendered by default in github so a browser cmd+F for "onContextChanged" didn't come up with anything. Here's where the change is: https://github.com/polynote/polynote/pull/77/files#r261373280

As for CSS, It was just moved here: https://github.com/polynote/polynote/pull/77/files#r261373537